daemon/networkdriver/portallocator/portallocator.go

```diff
@@ -59,6 +69,28 @@ func NewErrPortAlreadyAllocated(ip string, port int) ErrPortAlreadyAllocated {
 	}
 }
 
+func getPortRange() (int, int) {
+	const param = "/proc/sys/net/ipv4/ip_local_port_range"
+	var (
+		start   = defaultPortRangeStart
+		end     = defaultPortRangeEnd
+		rePorts = regexp.MustCompile(`\d+`)
+	)
+
+	if line, err := ioutil.ReadFile(param); err != nil {
+		log.Errorf("Failed to read %s kernel parameter: %s", param, err.Error())
+	} else {
+		ports := rePorts.FindAllString(string(line), 2)
+		start, _ = strconv.Atoi(ports[0])
```

Comment:
If that file will be empty there will be panic. You should check that `len(ports) == 2` and log error if not.